src/web/components/SlotBodyEnd/BrazeEpic.tsx

```diff
@@ -0,0 +1,169 @@
+import * as emotionTheming from 'emotion-theming';
+import {
+	parseBrazeEpicParams,
+	EpicDataFromBraze,
+	Variant,
+} from '@root/src/web/lib/braze/parseBrazeEpicParams';
+import { BrazeMessagesInterface } from '@root/src/web/lib/braze/BrazeMessages';
+import { CanShowResult } from '@root/src/web/lib/messagePicker';
+
+const { css } = emotion;
+
+const wrapperMargins = css`
+	margin: 18px 0;
+`;
+
+const EPIC_COMPONENT_PATH = '/epic.js';
+
+type NoEpicDataFromBraze = {
+	componentName: 'NoEpic';
+};
+
+type DataFromBraze = EpicDataFromBraze | NoEpicDataFromBraze;
+
+type Meta = {
+	dataFromBraze?: DataFromBraze;
+	logImpressionWithBraze: () => void;
+	// logButtonClickWithBraze: (id: number) => void; // TODO: Handle button click tracking
+};
+
+type EpicConfig = {
+	meta: Meta;
+	contributionsServiceUrl: string;
+	countryCode: string;
+};
+
+type Tracking = {
+	ophanPageId: string;
+	platformId: string;
+	clientName: string;
+	referrerUrl: string;
+};
+
+type EpicProps = {
+	variant: Variant;
+	tracking: Tracking;
+	numArticles: number;
+	countryCode: string;
+};
+
+export const canShow = async (
+	brazeMessagesPromise: Promise<BrazeMessagesInterface>,
+): Promise<CanShowResult> => {
+	try {
+		const brazeMessages = await brazeMessagesPromise;
+		const message = await brazeMessages.getMessageForEndOfArticle();
+
+		return {
+			result: true,
+			meta: {
+				dataFromBraze: message.extras,
+				logImpressionWithBraze: () => {
+					message.logImpression();
+				},
+			},
+		};
+	} catch (e) {
+		return { result: false };
+	}
+};
+
+const buildEpicProps = (
+	dataFromBraze: EpicDataFromBraze,
+	countryCode: string,
+): EpicProps | null => {
+	const variant = parseBrazeEpicParams(dataFromBraze);
+	if (variant === null) return null;
+
+	const pageTracking = {
+		ophanPageId: window.guardian.config.ophan.pageViewId,
+		platformId: 'GUARDIAN_WEB',
+		clientName: 'dcr',
+		referrerUrl: window.location.origin + window.location.pathname,
+	};
+
+	const tracking = {
+		...pageTracking,
+	};
+
+	return { variant, tracking, numArticles: 0, countryCode };
+};
+
+const BrazeEpic = ({
+	contributionsServiceUrl,
+	meta,
+	countryCode,
+}: EpicConfig) => {
+	const [Epic, setEpic] = useState<React.FC<EpicProps>>();
+
+	useEffect(() => {
+		window.guardian.automat = {
+			react: React,
+			preact: React,
+			emotionCore,
+			emotionTheming,
+			emotion,
+		};
+		const componentUrl = `${contributionsServiceUrl}${EPIC_COMPONENT_PATH}`;
+		window
+			.guardianPolyfilledImport(componentUrl)
+			.then((epicModule: { ContributionsEpic: React.FC<EpicProps> }) => {
+				setEpic(() => epicModule.ContributionsEpic); // useState requires functions to be wrapped
+				// TODO: log the impression when the epic is in view (using hasBeenSeen probably)
+				meta.logImpressionWithBraze();
+			})
+			// eslint-disable-next-line no-console
+			.catch((error) => console.log(`epic - error is: ${error}`));
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Usually, this lint rule is right - what was the reason for disabling it

Reply:
Ah good spot, yeah I think we can fill out the dependencies correctly and remove this commit.

Reply:
Changing this reminded me that the reason for `[]` (and therefore the linting rule exception) was because we only want/need this to happen once. I've replaced with the true dependency list and `useOnce` which I think makes things clearer.

Reply:
It's great to see this `useOnce` api working out!